cmd/kyma/provision/gardener/aws/cmd.go

```diff
@@ -0,0 +1,201 @@
+
+		RunE: func(_ *cobra.Command, _ []string) error { return c.Run() },
+	}
+
+	cmd.Flags().StringVarP(&o.Name, "name", "n", "", "Name of the cluster to provision. (required)")
+	cmd.Flags().StringVarP(&o.Project, "project", "p", "", "Name of the Gardener project where you provision the cluster. (required)")
+	cmd.Flags().StringVarP(&o.CredentialsFile, "credentials", "c", "", "Path to the kubeconfig file of the Gardener service account for a target provider. (required)")
+	cmd.Flags().StringVarP(&o.Secret, "secret", "s", "", "Name of the Gardener secret used to access the target provider. (required)")
+	cmd.Flags().StringVarP(&o.KubernetesVersion, "kube-version", "k", "1.16", "Kubernetes version of the cluster.")
+	cmd.Flags().StringVarP(&o.Region, "region", "r", "eu-west-3", "Region of the cluster.")
+	cmd.Flags().StringSliceVarP(&o.Zones, "zones", "z", []string{"eu-west-3a"}, "Zones specify availability zones that are used to evenly distribute the worker pool. eg. --zones=\"europe-west3-a,europe-west3-b\"")
+	cmd.Flags().StringVarP(&o.MachineType, "type", "t", "m5.xlarge", "Machine type used for the cluster.")
+	cmd.Flags().StringVar(&o.CIDR, "cidr", "10.250.0.0/16", "Gardener Classless Inter-Domain Routing (CIDR) used for the cluster.")
+	cmd.Flags().StringVar(&o.DiskType, "disk-type", "gp2", "Type of disk to use on the target provider.")
+	cmd.Flags().StringVar(&o.WCIDR, "workercidr", "10.250.0.0/16", "Specifies Gardener Classless Inter-Domain Routing (CIDR) of the workers of the cluster.")
+	cmd.Flags().IntVar(&o.DiskSizeGB, "disk-size", 30, "Disk size (in GB) of the cluster.")
+	cmd.Flags().IntVar(&o.NodeCount, "nodes", 3, `Number of cluster nodes which also defines the maximum autoscale value for the cluster. The value you provide for this flag overrides the settings defined with the "scaler-max" flag.`)
+	cmd.Flags().IntVar(&o.ScalerMin, "scaler-min", 2, "Minimum autoscale value of the cluster.")
+	cmd.Flags().IntVar(&o.ScalerMax, "scaler-max", 3, `Maximum autoscale value of the cluster. If you use the "nodes" flag in parallel to this one, it will result in overriding the "scaler-max" flag.`)
+	cmd.Flags().IntVar(&o.Surge, "surge", 3, "Maximum surge of the cluster.")
+	cmd.Flags().IntVarP(&o.Unavailable, "unavailable", "u", 1, "Maximum allowed number of unavailable nodes.")
+	cmd.Flags().StringVar(&o.NetworkType, "network-type", "calico", "Network type to be used.")
+	cmd.Flags().StringVar(&o.NetworkNodes, "network-nodes", "", "CIDR of the entire node network.")
+	cmd.Flags().StringVar(&o.NetworkPods, "network-pods", "", "Network type to be used.")
+	cmd.Flags().StringVar(&o.NetworkServices, "network-services", "", "CIDR of the service network.")
+	cmd.Flags().StringVar(&o.MachineImageName, "machine-image-name", "coreos", "Version of the shoot's machine image name in any environment.")
+	cmd.Flags().StringVar(&o.MachineImageVersion, "machine-image-version", "2303.3.0", "Version of the shoot's machine image version in any environment.")
```

Comment:
The network and machine image.. shouldn't it be maybe handled just as "extra" config so that we don't have to maintain that attributes here. I'm afraid of the default like for the machine-image-version which might change frequently and a user rarely wants to specify it at all..
Please compare with AKS and GKE and try to keep it consistent, what is extra, what not.

Reply:
I removed those flags. User can pass them as extra if they like.